core/plugins/org.polarsys.capella.core.libraries.ui/src/org/polarsys/capella/core/libraries/ui/handlers/ManageReferencedLibrariesHandler.java

```diff
@@ -79,20 +92,27 @@ protected Session getExistingSession(IFile file) {
   }
 
   @Override
-  public void removeHandlerListener(IHandlerListener handlerListener_p) {
-    super.removeHandlerListener(handlerListener_p);
+  public void removeHandlerListener(IHandlerListener handlerListener) {
+    super.removeHandlerListener(handlerListener);
   }
 
   @Override
-  public Object execute(ExecutionEvent event_p) throws ExecutionException {
-    TreeSelection selection = (TreeSelection) HandlerUtil.getCurrentSelection(event_p);
+  public Object execute(ExecutionEvent event) throws ExecutionException {
+    TreeSelection selection = (TreeSelection) HandlerUtil.getCurrentSelection(event);
     IFile file = (IFile) selection.getFirstElement();
     Session session = getExistingSession(file);
 
     IModel rootModel = ILibraryManager.INSTANCE.getModel(session.getTransactionalEditingDomain());
     if ((rootModel == null) || !(rootModel instanceof IModel.Edit)) {
       return null;
     }
+    
+    IStatus status = checkReferencedLibraries(rootModel, session);
+    if (!status.isOK()) {
+      MessageDialog.openError(Display.getCurrent().getActiveShell(), "Error", status.getMessage());
+      return null;
+    }
+    
```

Comment:
I think there might be a better way to do this.
I updated the original ticket with the details and some images :wink: